File: windows-ui/GSMRelayDesktop/Models/Relay.cs
using System.Collections.Generic;

namespace GSMRelayDesktop.Models;

public class Relay
{
    public long Id { get; set; }
    public string Name { get; set; } = "";
    public string PhoneNumber { get; set; } = "";
    public string Password { get; set; } = "";
    public string Location { get; set; } = "";
    public List<RelayUser> Users { get; set; } = new();
    public long? LastSync { get; set; }
    public bool CloudBackup { get; set; }

    public string DisplayName => string.IsNullOrWhiteSpace(Name) ? PhoneNumber : Name;
}

File: android/app/src/main/java/com/security/gsmrelay/model/Relay.kt
package com.security.gsmrelay.model

data class Relay(
    val id: Long,
    val name: String,
    val phoneNumber: String,
    val password: String,
    val location: String = "",
    val users: List<User> = emptyList(),
    val lastSync: Long? = null,
    val cloudBackup: Boolean = false
)
